### ISSUE TITLE
Pristine form not validating before submit is called

### ISSUE THREAD

**Issue by RyanAtViceSoftware** [erikras/redux-form#1862]
If my form is dirty then submit won't be called if it's got errors but if it's pristine then submit is called and the form is submitted. Am I doing something wrong or is this a bug?

Note I created this question on stack overflow in the event that this is just me doing something wrong: http://stackoverflow.com/questions/39693856/is-there-a-way-to-make-a-form-validate-from-code

**Comment by Aaronius** [comment]
Possibly related: #1477.

**Comment by bertjwregeer** [comment]
As a workaround, if you have a sync validate function, call it in your submit function, and throw `SubmissionError` with the returned info from the validate function.

```
function _validate(state, props) {
    return {field: ['error']};
}
```

And then your onSubmit function:

```
    onSubmit = (state) => {
        const { dispatch } = this.props;

        return new Promise((resolve, _reject) => {
            const errors = _validate(state, {});

            if (!Object.keys(errors).every((val) => { return errors[val] === null; })) {
                throw new SubmissionError({
                    ...errors,
                    _error: 'Form could not be submitted.',
                });
            }

            dispatch(...);
        });
    }
```

Then:

```
@reduxForm({ form: 'formname', validate: _validate })
```

Now you have something stopping submission, and adding a bunch of errors to the form.

**Comment by RyanAtViceSoftware** [comment]
@bertjwregeer thanks, I'll give that a try

**Comment by tepamid** [comment]
@RyanAtViceSoftware the issue seems to be fixed in reduxForm 7.0.3

**Comment by danielrob** [comment]
@tepamid thanks and indeed! Though note issue 3284 :). Closing.
